subseas_flex_fcst/maproom.py

```diff
@@ -212,7 +248,10 @@ def local_plots(marker_pos, start_date, lead_time):
 
     # Forecast CDF
     fcst_q, fcst_mu = xr.broadcast(quantiles, fcst_mu)
-    fcst_dof = int(fcst_var.attrs["dof"]) #int(dofVar)
+    try:
+        fcst_dof = int(fcst_var.attrs["dof"])
+    except:
+        fcst_dof = obs["T"].size - 1
```

Comment:
I'm wondering if it's really a good idea to silently default to a fallback value rather than raising an exception because the dof metadata were missing. Under what circumstances would it not be given?

Reply:
Historically, it hasn't been given. It's somewhat recent that it has been written into the files. But I guess not recent enough for those Guatemala ACToday forecasts not to have it...

Reply:
We can't figure out the correct value. It depends on the number of modes retained by CCA (if CCA used, if not, other considerations) for the predictor and predictand. If NextGen, there are cascading CCAs (models first then their combination). I was given a formula for that case that ended up having negative dofs (or ridiculously low) in some cases so I stepped back from that. Could not get answers from forecasters. And even if we had the right calculations, I am not sure if we have more changed to get the number of modes or the dof been given.
And the correct value is going to depend on month-of-S and L, so if you're thinking of putting dof in the config or something, it's actually going to have to be a DataArray.
So the occasion will be increasingly rare but incredibly painful to get right.

My take anyway is that it should not be the app's job to tells us whether the dof is missing or not, but rather whatever libraries are going to curate data properly in the first place. A somewhat not helpful comment since all is done together at present.

But Ingrid has used dof = n-1 for years, Simon knowing it, without raising many eyebrows (mostly raising my perfectionist ones), so at the stage we're at, it might be just fine a it is.

Reply:
OK. If there's nothing we can do about it then there's no point raising an exception.